ranger/core/loader.py

```diff
@@ -51,11 +51,11 @@ def destroy(self):
 class CopyLoader(Loadable, FileManagerAware):  # pylint: disable=too-many-instance-attributes
     progressbar_supported = True
 
-    def __init__(self, copy_buffer, do_cut=False, overwrite=False):
+    def __init__(self, copy_buffer, do_cut=False, overwrite=False, dest=None):
         self.copy_buffer = tuple(copy_buffer)
         self.do_cut = do_cut
         self.original_copy_buffer = copy_buffer
-        self.original_path = self.fm.thistab.path
+        self.original_path = dest if dest is not None else self.fm.thistab.path
```

Comment:
I'm wondering if we could not just define the default for `dest=self.fm.thistab.path`?

Reply:
I don't think it'd work with a "dynamic" value as they are only assigned *once*. (that's also the reason why a mutable default is a very bad idea most of the time)

Reply:
Yep, probably not a good idea. I've just read [this](https://docs.python-guide.org/writing/gotchas/) and I think what it will happen is that the default `dest? will always be the directory of the first tab in ranger.

Reply:
Thanks for pointing this out, @vifon.

@palikar, the current failure in travis is because pylint found some trailing whitespace somewhere. We're strict about applying pep8 and pylint, so even if the tests pass you'll have to pass the linting as well. Make sure you have all the testing dependencies, flake8, pylint, etc.

Reply:
Ok, I cleared up the code and all checks on the Travis build now pass.